gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js

```diff
@@ -35,45 +43,197 @@ fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource", {
+ *          the OS version provided.
+ * @return {Promise} A promise that will be resolved with results (see above) or
+ *          rejected on error.
+ */
+gpii.flowManager.solutionsRegistry.dataSource.cloud.handle = function (that, requestOptions) {
+    var promise = fluid.promise();
+    if (requestOptions.os && requestOptions.os in that.fullSolutionsRegistry) {
+        promise.resolve(that.fullSolutionsRegistry[requestOptions.os]);
+    } else {
+        promise.reject({
+            isError: true,
+            message: "The requested OS (" + requestOptions.os + ") was not present in the solutions registry",
+            statusCode: 404
+        });
+    }
+    return promise;
+};
+
+// Solutions registry datasource used by the Local Flow Manager.  It loads both
+// solutions from the source code repository as well as the local file sysmtem.
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource.local", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester"
+        },
+        repositorySolutionsLoader: {
+            type: "gpii.flowManager.repositorySolutionsLoader"
+        }
+        // The platformReporter subcomponent is distributed down from local
+        // flowManager
+//      platformReporter: {
+//          type: "gpii.platformReporter.native"
+//      }
+    },
+    readOnlyGrade: "gpii.flowManager.solutionsRegistry.dataSource.local",
+    members: {
+        repositorySolutionsRegistry: null
+    },
+    invokers: {
+        getImpl: {
+            funcName: "gpii.flowManager.solutionsRegistry.dataSource.local.handle",
+            args: ["{that}", "{arguments}.1"]
+                             // options
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
+            listener: "gpii.flowManager.solutionsRegistry.dataSource.local.loadFromRepository",
+            args: ["{that}", "{arguments}.0", "{that}.platformReporter"],
+                             // revision
+            priority: "after:getRevision"
+        },
+        "loadSolutions.solutionsLoaded": {
+            listener: "{that}.events.solutionsRegistryReady",
+            priority: "after:loadFromRepository"
+        }
+    }
+});
+
+/**
+ * Retrieve the solutions registry JSON file from the source code repository.
+ * If successful, sets the repositorySolutionsRegistry member to the fetched
+ * solutions registry.
+ * @param {Component} that - An instance of gpii.flowManager.solutionsRegistry.dataSource
+ * @param {Component} that.repositorySolutionsLoader - An instance of
```

Comment:
The parameters described at line 189 and 190 are not in the argument list. These descriptions can be removed.

Reply:
Done.